Lib/multiprocessing/popen_fork.py

```diff
@@ -70,7 +70,7 @@ def _launch(self, process_obj):
         if self.pid == 0:
             try:
                 os.close(parent_r)
-                code = process_obj._bootstrap()
+                code = process_obj._bootstrap(parent_sentinel=child_w)
```

Comment:
I don't understand. `child_w` will always be "ready" (writable) except if the pipe buffer is full, so how can you use it as a sentinel?

Reply:
The idea is to use `_ParentProcess.is_alive`, that itself calls `mutliprocessing.connection.wait` on the sentinel. Does that answer your question?

Reply:
Not really, because it doesn't answer how it works concretely on the write end of the pipe.
```
>>> import os, select
>>> r, w = os.pipe()
>>> select.select([r], [r], [r])
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyboardInterrupt
>>> select.select([w], [w], [w])
([], [4], [])
```

Reply:
Though, at least on Linux:
```
>>> select.select([w], [], [w])
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyboardInterrupt
>>> os.close(r)
>>> select.select([w], [], [w])
([4], [], [])
```

Reply:
Actually, I think we should use `child_r` and not `child_w`, as we do for `spawn`. But actually @tomMoral wrote this line, maybe I am missing something?

Reply:
I am not a frequent user of `select`, but it does seem odd to have a write end of a pipe into the `rlist` of `select` (which is the behavior we rely on using `multiprocessing.connection.wait`). But actually, I did not write this line, @tomMoral did. Maybe he has other insights on this. 
@pitrou would you be more confortable if we use the read end of a new pipe as a sentinel instead to mimicate the `sentinel` implementation of the parent process for `fork`?

Reply:
Yes, that would seem like a more strictly POSIX-compliant solution.

Reply:
Yes indeed, it seems weird that way. I did not realize this was different compared to `spawn_posix`, where you have access to `child_r`. (But there seems to be a leak with `parent_w` with this implem, cf other comment).

That being said, the goal of the sentinel is simply to know if this pipe is still open or not. That is why it feels a waste to create a new pipe. But there might not be better answer.